src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -280,16 +286,22 @@ float64_t CCrossValidation::evaluate_one_run()
 			/* set feature subset for training */
 			SGVector<index_t> inverse_subset_indices=
 					m_splitting_strategy->generate_subset_inverse(i);
-			m_features->add_subset(inverse_subset_indices);
+
+			CFeatures* features=CFeatures::create_shallow_copy(m_features);
```

Comment:
I see. The problem with cloning the features by hand is that *if* they had already been assigned to the machine in some way (via kernel e.g.), then they *will* already have been cloned. But OK, let's postpone that as well.
Can you add all these things to a TODO list so that we wont forget?